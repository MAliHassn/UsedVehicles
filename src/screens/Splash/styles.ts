import {StyleSheet} from 'react-native';
import {useCustomTheme} from '../../constants/useCustomTheme';
import {SCREEN_HEIGHT, SCREEN_WIDTH} from '../../constants/Screen';

const useStyles = () => {
  const {colors, fonts} = useCustomTheme();
  const styles = StyleSheet.create({
    container: {
      flex: 1,
      alignItems: 'center',
      justifyContent: 'center',
      backgroundColor: colors.white,
    },
    image: {
      aspectRatio: 2,
      height: 'auto',
      width: SCREEN_WIDTH * 0.7,
    },
    bottom_view: {
      position: 'absolute',
      bottom: 10,
      flexDirection: 'row',
      justifyContent: 'center',
      alignItems: 'center',
    },
    text: {
      color: colors.black,
      fontSize: fonts.body,
    },
    rak_bank_image: {
      aspectRatio: 1,
      height: 'auto',
      width: SCREEN_WIDTH * 0.15,
    },
    bottomBg: {
      position: 'absolute',
      bottom: 30,
      width: SCREEN_WIDTH,
      height: SCREEN_HEIGHT / 2.5,
      zIndex: -1,
    },
  });

  return {
    styles,
  };
};

export default useStyles;
