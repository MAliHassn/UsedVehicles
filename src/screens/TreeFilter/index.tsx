export { default as TreeFilter } from './TreeFilter';