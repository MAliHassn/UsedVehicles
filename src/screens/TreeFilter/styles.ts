import {StyleSheet} from 'react-native';
import {useCustomTheme} from '../../constants/useCustomTheme';

const useStyles = () => {
  const {colors, fonts} = useCustomTheme();
  const styles = StyleSheet.create({
    container: {
      flex: 1,
      backgroundColor: colors.white,
    },
    filter: {
      padding: 10,
      marginBottom: 110,
      paddingTop: 0,
      flex: 1,
    },
    selected: {
      justifyContent: 'center',
      alignItems: 'center',
      position: 'absolute',
      bottom: 80,
      elevation: 8,
      left: 0,
      right: 0,
      height: 40,
      marginHorizontal: 10,
    },
    apply: {
      justifyContent: 'center',
      alignItems: 'center',
      position: 'absolute',
      bottom: 20,
      elevation: 8,
      left: 0,
      right: 0,
      height: 50,
      marginHorizontal: 10,
    },
    row: {
      flexDirection: 'row',
      justifyContent: 'center',
    },
    text: {
      color: colors.white,
      fontSize: fonts.heading,
    },
    subtext: {
      color: colors.black,
      fontSize: fonts.sub_body,
    },
  });

  return {
    styles,
  };
};

export default useStyles;
