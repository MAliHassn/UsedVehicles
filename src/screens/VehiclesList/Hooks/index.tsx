export { default as useVehicles } from './useVehicles';
