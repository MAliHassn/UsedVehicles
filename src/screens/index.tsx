export { Splash } from './Splash';
export { VehiclesList } from './VehiclesList';
export { TreeFilter } from './TreeFilter';
