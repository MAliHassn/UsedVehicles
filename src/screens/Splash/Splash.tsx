import { Image, StyleSheet, Text, View } from 'react-native';
import React from 'react';
import { Colors } from '../../constants/Colors';
import { images } from '../../constants/Images';
import { SCREEN_HEIGHT, SCREEN_WIDTH } from '../../constants/Screen';
import { Fonts } from '../../constants/Fonts';
import {useSplash} from './Hooks';

const Splash = () => {
  const {} = useSplash();

  return (
    <View style={styles.container}>
      <Image source={images.app_logo} style={styles.image} resizeMethod="auto" />

      <View style={styles.bottom_view}>
        <Text style={styles.text}>Powered by</Text>
        <Image source={images.rak_bank} style={styles.rak_bank_image} resizeMethod="auto" />
      </View>

      <Image
        source={images.splashBg}
        style={styles.bottomBg}
        resizeMode="stretch"
      />

    </View>
  );
};

export default Splash;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: Colors.white
  },
  image: {
    aspectRatio: 2,
    height: "auto",
    width: SCREEN_WIDTH * 0.7,
  },
  bottom_view: {
    position: "absolute",
    bottom: 10,
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center"
  },
  text: {
    color: Colors.black,
    fontSize: Fonts.body
  },
  rak_bank_image: {
    aspectRatio: 1,
    height: "auto",
    width: SCREEN_WIDTH * 0.15,
  },
  bottomBg: {
    position: 'absolute',
    bottom: 30,
    width: SCREEN_WIDTH,
    height:SCREEN_HEIGHT/2.5,
    zIndex:-1
  },
});