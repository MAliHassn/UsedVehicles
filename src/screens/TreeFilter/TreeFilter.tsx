import { Pressable, StyleSheet, Text, View } from 'react-native';
import React from 'react';
import { Colors } from '../../constants/Colors';
import { Fonts } from '../../constants/Fonts';
import Header from '../../components/Header';
import Filter from '../../components/Filter';
import LinearGradient from 'react-native-linear-gradient';
import { useFilter } from './Hooks';

const TreeFilter = () => {
  const { realTimeChanges,
    clickOnNodes,
    filter_data,
    checkedNode,
    checkedChilds,
    returnRow
  } = useFilter()

  return (
    <View style={styles.container}>

      <Header />

      <View style={styles.filter}>
        <Filter
          key={"test"}
          data={filter_data as Node[]}
          onPress={(v) => { realTimeChanges(v) }}
          onPressNode={(v) => { clickOnNodes(v) }}
        />
      </View>

      <LinearGradient
        colors={["rgba(0,0,0,0.3)", "rgba(0,0,0,0.2)",]}
        style={styles.selected}
        start={{ x: 1, y: 0 }}
        end={{ x: 0.2, y: 0.9 }}
      >

        {returnRow(
          "tree ids:",
          checkedChilds?.ids,
          checkedNode?.id
        )}
        {returnRow(
          "tree titles:",
          checkedChilds?.titles,
          checkedNode?.title
        )}

      </LinearGradient>

      <LinearGradient
        colors={[Colors.secondary, Colors.primary,]}
        style={styles.apply}
        start={{ x: 1, y: 0 }}
        end={{ x: 0.2, y: 0.9 }}
      >
        <Pressable onPress={() => { console.log("") }}>
          <Text style={styles.text}>{"APPLY"}</Text>
        </Pressable>
      </LinearGradient>

    </View>
  );
};

export default TreeFilter;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.white
  },
  filter: {
    padding: 10,
    marginBottom: 110,
    paddingTop: 0,
    flex: 1
  },
  selected: {
    justifyContent: "center",
    alignItems: "center",
    position: "absolute",
    bottom: 70,
    elevation: 8,
    left: 0,
    right: 0,
    height: 40,
    marginHorizontal: 10,
  },
  apply: {
    justifyContent: "center",
    alignItems: "center",
    position: "absolute",
    bottom: 10,
    elevation: 8,
    left: 0,
    right: 0,
    height: 50,
    marginHorizontal: 10
  },
  row: {
    flexDirection: "row",
    justifyContent: "center"
  },
  text: {
    color: Colors.white,
    fontSize: Fonts.heading
  },
  subtext: {
    color: Colors.black,
    fontSize: Fonts.sub_body
  },

});