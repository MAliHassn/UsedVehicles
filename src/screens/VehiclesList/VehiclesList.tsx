import { Image, StyleSheet, View, TextInput, TouchableOpacity, FlatList, ListRenderItemInfo, } from 'react-native';
import React, { useCallback, useRef } from 'react';
import { Colors } from '../../constants/Colors';
import { Fonts } from '../../constants/Fonts';
import VehiclesListItem from '../../components/VehiclesListItem';
import TabsSwitcher from '../../components/TabsSwitcher';
import { images } from '../../constants/Images';
import LinearGradient from 'react-native-linear-gradient';
import { useVehicles } from "./Hooks"

const VehiclesList = () => {
  const listRef = useRef<FlatList>(null);
  const {
    navigateTo,
    filterUpdate,
    mainFilters,
    viewConfigRef,
    callBackUpdate,
    vehiclesData,
    input,
  } = useVehicles()

  const renderItem = useCallback(
    ({ item, index }: ListRenderItemInfo<IVehiclesItem>) => {
      return <VehiclesListItem key={index.toString()} item={item} callBackFav={() => {
        console.log("pressed")
      }} />;
    }, []);


  return (
    <View style={styles.container}>

      <LinearGradient
        colors={[Colors.secondary, Colors.primary,]}
        style={styles.linerar}
        start={{ x: 1, y: 0 }} end={{ x: 0.2, y: 0.9 }}
      >
        <View style={styles.viewSearch}>

          <View style={{ flex: 0.15, elevation: 5, }}>
            <TouchableOpacity
              style={styles.icon}
              onPress={() => {
                if (!isNaN(Number(input)) && Number(input) > -1) {
                  callBackUpdate(input);
                } else {
                  return;
                }
              }}>
              <Image source={images.search} style={styles.image} resizeMethod="auto" />
            </TouchableOpacity>
          </View>

          <View style={{ flex: 0.7, paddingStart: 5 }}>
            <TextInput
              value={input}
              onSubmitEditing={() => callBackUpdate(input)}
              onChangeText={(text) => callBackUpdate(text)}
              style={styles.input}
              placeholder={"Search in vehicles"}
              placeholderTextColor={Colors.gray}
            />
          </View>

          <View style={styles.filter}>
            <TouchableOpacity
              testID='filter'
              style={styles.icon}
              onPress={() => {
                navigateTo()
                // navigation.navigate('TreeFilter')
              }}>
              <Image source={images.filter} style={styles.image} resizeMethod="auto" />
            </TouchableOpacity>
          </View>

        </View>
      </LinearGradient>

      <View style={{ marginVertical: 10, alignSelf: "flex-start" }}>
        <TabsSwitcher
          key={"tabs"}
          data={mainFilters}
          callBackSecTabSwitch={(item: any, index: number) => {
            filterUpdate(item)
          }}
        />
      </View>

      <FlatList
        ref={listRef}
        disableVirtualization={true}
        showsVerticalScrollIndicator={false}
        removeClippedSubviews
        viewabilityConfig={viewConfigRef.current}
        contentContainerStyle={{ paddingBottom: 20, }}
        data={vehiclesData as IVehiclesItem[]}
        numColumns={2}
        keyExtractor={(item, index) => index.toString()}
        renderItem={renderItem}
      />

    </View>
  );
};

export default VehiclesList;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Colors.white
  },

  viewSearch: {
    padding: 5,
    elevation: 10,
    flexDirection: 'row',
    borderRadius: 50,
    borderWidth: 1,
    borderColor: Colors.white,
    height: 40
  },
  input: {
    flex: 1,
    padding: 5,
    paddingLeft: 5,
    fontSize: Fonts.sub_heading,
    height: "auto",
    color: Colors.white
  },
  icon: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
  },
  image: {
    width: 22,
    height: 22
  },
  linerar: {
    padding: 10,
    justifyContent: "center",
    // alignItems: "center",
    elevation: 8,
  },
  filter: {
    flex: 0.15,
    elevation: 5,
    justifyContent: "center",
    alignItems: "center"
  },

});