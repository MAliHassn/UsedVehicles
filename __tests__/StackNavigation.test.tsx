import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { render, screen, fireEvent } from '@testing-library/react-native';
import {Splash} from '../src/screens/Splash';
import {TreeFilter} from '../src/screens/TreeFilter';
import {VehiclesList} from '../src/screens/VehiclesList';

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

describe('Testing react navigation', () => {
  test('page contains 3 images', async () => {
    const component = (
      <NavigationContainer>
        <Splash />
      </NavigationContainer>
    );

    render(component);
    expect(render).toMatchSnapshot();
  })

  test('page contains list of vehicles data ', async () => {
    const component = (
      <NavigationContainer>
        <VehiclesList />
      </NavigationContainer>
    );

    render(component);
    expect(render).toMatchSnapshot();
  })

  test('page contains tree filter ', async () => {
    const component = (
      <NavigationContainer>
        <TreeFilter />
      </NavigationContainer>
    );

    render(component);
    expect(render).toMatchSnapshot();
  })
});