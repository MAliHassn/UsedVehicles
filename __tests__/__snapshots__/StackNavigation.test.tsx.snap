// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing react navigation page contains 3 images 1`] = `[Function]`;

exports[`Testing react navigation page contains list of vehicles data  1`] = `[Function]`;

exports[`Testing react navigation page contains tree filter  1`] = `[Function]`;
